lib/ruby_indexer/lib/ruby_indexer/reference_finder.rb

```diff
@@ -5,6 +5,35 @@ module RubyIndexer
   class ReferenceFinder
     extend T::Sig
 
+    class Target
+    end
+
+    class ConstTarget < Target
+      extend T::Sig
+
+      sig { params(fully_qualified_name: String).void }
+      def initialize(fully_qualified_name)
+        super()
+        @fully_qualified_name = fully_qualified_name
+      end
+
+      sig { returns(String) }
+      attr_reader :fully_qualified_name
+    end
+
+    class MethodTarget < Target
+      extend T::Sig
+
+      sig { params(method_name: String).void }
+      def initialize(method_name)
+        super()
+        @method_name = method_name
+      end
+
+      sig { returns(String) }
+      attr_reader :method_name
```

Comment:
NIT: Move the attributes above the regular method declarations.

Reply:
Do we really need a cop for this though?

Reply:
If it's a style decision we care about, and something that trivially handled by an exising cop, then I don't see why not.